BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
++
++    // Waterfall start - we're going on an adventure!
++    public Component permissionMessage() {
++        return permissionMessage;
++    }
++
++    protected void permissionMessage(final Component permissionMessage) {
++        this.permissionMessage = permissionMessage;
++    }
++    // Waterfall end
+ }
+diff --git a/api/src/main/java/net/md_5/bungee/api/score/Objective.java b/api/src/main/java/net/md_5/bungee/api/score/Objective.java
+index 0dd25646..f96ef1e4 100644
+--- a/api/src/main/java/net/md_5/bungee/api/score/Objective.java
++++ b/api/src/main/java/net/md_5/bungee/api/score/Objective.java
+@@ -2,6 +2,7 @@ package net.md_5.bungee.api.score;
+ 
+ import lombok.AllArgsConstructor;
+ import lombok.Data;
++import net.kyori.adventure.text.Component; // Waterfall - we're going on an adventure!
+ 
+ /**
+  * Represents an objective entry.
+@@ -15,6 +16,12 @@ public class Objective
+      * Name of the objective.
+      */
+     private final String name;
++    // Waterfall start - we're going on an adventure!
++    /**
++     * Display name of the objective.
++     */
++    private final Component displayName;
++    // Waterfall end
+     /**
+      * Value of the objective.
+      */
+@@ -23,4 +30,10 @@ public class Objective
+      * Type; integer or hearts
+      */
+     private String type;
++
++    // Waterfall start - write previously Lombok-generated constructor - we're going on an adventure!
++    public Objective(final String name, final String value, final String type) {
++        this(name, Component.empty(), value, type);
++    }
++    // Waterfall end
+ }
+diff --git a/api/src/main/java/net/md_5/bungee/api/score/Team.java b/api/src/main/java/net/md_5/bungee/api/score/Team.java
+index 849ba1cf..db3f95b0 100644
+--- a/api/src/main/java/net/md_5/bungee/api/score/Team.java
++++ b/api/src/main/java/net/md_5/bungee/api/score/Team.java
+@@ -6,6 +6,10 @@ import java.util.HashSet;
+ import java.util.Set;
+ import lombok.Data;
+ import lombok.NonNull;
++// Waterfall start - we're going on an adventure!
++import net.kyori.adventure.text.Component;
++import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
++// Waterfall end
+ 
+ @Data
+ public class Team
+@@ -13,9 +17,14 @@ public class Team
+ 
+     @NonNull
+     private final String name;
+-    private String displayName;
+-    private String prefix;
+-    private String suffix;
++    // Waterfall start - we're going on an adventure!
++//    private String displayName;
++    private Component displayName;
++//    private String prefix;
++    private Component prefix;
++//    private String suffix;
++    private Component suffix;
++    // Waterfall end
+     private byte friendlyFire;
+     private String nameTagVisibility;
+     private String collisionRule;
+@@ -36,4 +45,62 @@ public class Team
+     {
+         players.remove( name );
+     }
++
++    // Waterfall start - add previously Lombok-generated getters/setters - we're going on an adventure!
```

Comment:
should probably remove the Data annotation in that case

Reply:
Would mean that we would have to manually rewrite all of the getters and setters, instead of just being able to manually write this one and leave the others.